scripts/lib/api/generateMdxComponents.ts

```diff
@@ -0,0 +1,361 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { CheerioAPI, Cheerio, Element } from "cheerio";
+import { unified } from "unified";
+import rehypeParse from "rehype-parse";
+import rehypeRemark from "rehype-remark";
+import remarkStringify from "remark-stringify";
+
+import { ApiType } from "./Metadata";
+import { getLastPartFromFullIdentifier } from "../stringUtils";
+
+export type componentProps = {
+  id: string;
+  name?: string;
+  attributeType?: string;
+  attributeValue?: string;
+  githubSourceLink?: string;
+  signature?: string;
+  extraSignatures?: string[];
+};
+
+export async function processMdxComponent(
+  $: CheerioAPI,
+  $main: Cheerio<any>,
+  signatures: Cheerio<Element>[],
+  $dl: Cheerio<any>,
+  priorApiType: ApiType | undefined,
+  apiType: ApiType,
+  id: string,
+): Promise<[string, string]> {
+  findByText($, $main, "em.property", apiType).remove();
+
+  const $firstSignature = signatures.shift()!;
+  const githubSourceLink = prepareGitHubLink(
+    $firstSignature,
+    apiType === "method",
+  );
+
+  const componentProps = prepareProps(
+    $,
+    $firstSignature,
+    $dl,
+    priorApiType,
+    apiType,
+    githubSourceLink,
+    id,
+  );
+
+  const extraProps = signatures
+    .map(($overloadedSignature) =>
+      prepareProps(
+        $,
+        $overloadedSignature,
+        $dl,
+        apiType,
+        apiType,
+        prepareGitHubLink($overloadedSignature, apiType === "method"),
+        id,
+      ),
+    )
+    .flatMap((prop) => (prop ? prop : []));
+
+  if (componentProps) {
```

Comment:
Properties and attributes can return without any props if there's no signature

Reply:
What does that mean? Will they not get a `<Attribute>` component?

Reply:
Yes. There's a case in attributes and properties where you could have empty props. I haven't seen any case, but our script allowed it before, so I maintained the option here to not have any unexpected behavior. If the attribute/property doesn't have an id, I think we shouldn't treat it as a component, wdyt?

Reply:
I'm a big fan of removing edge cases when possible. If we haven't seen the edge case before with over 60 docs versions, we're unlikely to see it now. So it would be helpful to fail eagerly. Run over our current docs with this eager error to verify we indeed never have the edge case.

Keep in mind that the original script might be overly generalized in some places because we didn't always have a perfect understanding of Sphinx, so we might have defensively added allowances that aren't actually necessary. Or, you've seen cases where there are edge cases we didn't know about.